net/dns/manager.go

```diff
@@ -147,20 +147,28 @@ func (m *Manager) setLocked(cfg Config) error {
 		ocfg.WriteToBufioWriter(w)
 	}))
 
-	if err := m.resolver.SetConfig(rcfg); err != nil {
-		return err
-	}
+	m.resolver.SetConfig(rcfg)
+
 	if err := m.os.SetDNS(ocfg); err != nil {
-		m.health.SetDNSOSHealth(err)
+		m.health.SetUnhealthy(OSConfigWarnable, health.Args{health.ArgError: err.Error()})
 		return err
 	}
 
-	m.health.SetDNSOSHealth(nil)
+	m.health.SetHealthy(OSConfigWarnable)
 	m.config = &cfg
 
 	return nil
 }
 
+var OSConfigWarnable = health.Register(&health.Warnable{
```

Comment:
Having this exported (ignoring the lack of docs :)) isn't ideal: it means the responsibility for who owns updating it to/from healthy isn't clear.

If the problem is that the OSConfigurator is an interface so can't be trusted to do this, perhaps what we need is to instead make a top-level func in this package whose job is to call SetDNS on an OSConfigurator and update a health tracker.

Like:

```go
// SetOSDNS calls c.SetDNS with cfg, updating ht in the process.
func SetOSDNS(c OSConfigurator, ht *health.Tracker, cfg Config) error {
```